src/Features/LanguageServer/Protocol/Handler/RequestContext.cs

```diff
@@ -116,6 +116,32 @@ namespace Microsoft.CodeAnalysis.LanguageServer.Handler;
                 throw new InvalidOperationException();
             }
 
+            return _lspSolution.Value.Document as Document;
```

Comment:
I think this should throw if we ask for a Document, but get a TextDocument.  The handler should be explicit in asking for what it needs.

Reply:
Done